rustls/src/client/tls12.rs

```diff
@@ -462,21 +463,12 @@ fn emit_clientkx(transcript: &mut HandshakeHash, common: &mut CommonState, pubke
 
 fn emit_certverify(
     transcript: &mut HandshakeHash,
-    client_auth: &mut ClientAuthDetails,
+    signer: &dyn Signer,
     common: &mut CommonState,
 ) -> Result<(), Error> {
-    let (signer, message) = match (client_auth.signer.take(), transcript.take_handshake_buf()) {
-        (Some(signer), Some(msg)) => (signer, msg),
-        (None, _) => {
-            trace!("Not sending CertificateVerify, no key");
-            transcript.abandon_client_auth();
-            return Ok(());
-        }
-        (_, None) => {
-            trace!("Not sending CertificateVerify, no transcript");
-            return Ok(());
-        }
-    };
+    let message = transcript
+        .take_handshake_buf()
+        .ok_or_else(|| Error::General("Expected transcript".to_owned()))?;
```

Comment:
This is a change in behavior in the case where we weren't able to get the transcript. I think that's an impossible case? Regardless, the old logic seems wrong.